src/intro.md

```diff
@@ -10,12 +10,38 @@ This is the official “LinuxBoot Book” for the LinuxBoot project. The book:
 
 ## What is LinuxBoot?
 
-LinuxBoot is a project that aims to replace specific [firmware](./glossary.md)
-functionality with a Linux kernel and runtime. Over the years this project has
-grown to include various initiatives with the overarching goal of moving from
-obscure, complex firmware to simpler, open source firmware.
-
-The goal of LinuxBoot is to reduce the role of firmware to a small,
+LinuxBoot is based on the idea of replacing proprietary or corporate-driven
+late-stage boot [firmware](./glossary.md) with the Linux kernel and a
+community-based userland. That idea grew into a project that over the years
+includes various initiatives with the overarching goal of moving from obscure
+and complex firmware to simpler and open source firmware.
+
+The LinuxBoot project provides two reference implementations; linuxboot and
+Heads. The [linuxboot](https://github.com/linuxboot/linuxboot) build system
+outputs a boot payload consisting of a Linux kernel and an
+[initramfs](https://de.wikipedia.org/wiki/Initramfs) that contains a Golang
+userland built using [u-root](https://github.com/u-root/u-root).
+
+The Heads build system is more focused on local attestation and uses BusyBox
+instead of Golang.
```

Comment:
@tlaurion since I saw your :+1: on Daniel's comment I figured you might have something to add here? :smile:

Reply:
I'm glad it's named, I would not try to define it but point to the project page instead.

Reply:
I would really like to provide some context of what Heads is. Currently, it is really hard to understand the OSFW ecosystem. My goal is to make it easier to understand the numerous projects and how they relate. How would you place Heads in the context of OSFW?

Reply:
I'm not good at this.

A draft, reminding that linuxboot is concept, more focusing on UEFI pruning and replacing DXE with u-root, where u-root could be replaced by Heads as it's Linux based payload. And where Heads focuses on being a coreboot payload, where historically, linuxboot contributed to Heads so that Heads could generate a linuxboot UEFI usable payload. Also, Heads provides complete externally flashable roms for all supported boards, where it's still not clear to me on which real physical boards linuxboot is used today, graveyard.md showing removal of past boards and reducing current boards to old, some now unablvailable boards. 


Anyway, a draft:

**LinuxBoot with u-root**:
- **u-root**: Provides a set of standard Unix tools rewritten in Go, such as `ls`, `cp`, `shutdown, etc.
- **Boot policy tools**: Includes tools for PXE booting, booting via GRUB config, and other boot-related policies.
- **Minimal initramfs**: Creates a single binary root file system (initramfs) containing BusyBox-like utilities.

**LinuxBoot with Heads**:
- **Heads**: Includes a comprehensive suite of standard Linux ecosystem tools for creating a full Linux recovery environment.
- **Extensive toolset**: Incorporates `qrencode`, `hotp-verification`, `io386`, `cryptsetup`, `gnupg`, `tpm toolstack`, `flashprog`, `lvm`, `e2fsprogs`, `extfsprogs`, and more.
- **Security features**: Focuses on local attestation, TPM DUK seal/unseal operations, GPG-based security measures, and reproducible builds.

In essence, while LinuxBoot with u-root offers a streamlined and flexible boot environment with basic Unix tools, LinuxBoot with Heads provides a robust and secure boot environment with a wide range of tools necessary for a complete Linux recovery setup.

Reply:
Also note présence of other projects, like tinyboot, ownerboot and tvix, all reusing or aiming to reuse linuxboot concept (Linux as payload/linux as bootloader replacing as much opaque/proprietary/unneeded drivers with Linux) which would do it better if there was more collaboration between those projects and working together in the goal of freeing motherboards of bloat/proprietary blobs/changing the ecosystem to be freer.


But to accomplish this, it needs to be easy to build, easy to deploy and supported by vendors with transparency on what to buy how to deploy and being reproducible with simple commands, maintained and sold by oems on their supported hardware, and ideally, pushed by ODM so users don't need to be hackers and understand everything to use it.

This is what Heads try to accomplish.

Reply:
Tldr: refer to project page instead of trying to define heads, which is criticized to do too much but where OEM requires more to deploy easily, securely, with proper transfer of ownership and in transit interdiction of hardware.

I have difficulties to define Heads myself, hence good job trying to define it.

A good article amde by unbiased third party is all I can refer to. 

- What is Heads: https://tech.michaelaltfield.net/2023/02/16/evil-maid-heads-pureboot/

Reply:
Also note https://www.linuxboot.org/page/faq/

LinuxBoot, NERF, HEADS? What’s What?

LinuxBoot is the project that replaces specific firmware functionality with a Linux kernel. LinuxBoot is agnostic to what initramfs is used with the kernel.

NERF is LinuxBoot with u-root as the initramfs. u-root contains boot policy tools in Go (e.g. PXE booting, booting via GRUB config) among standard busybox-like utilities rewritten in Go.

> HEADS is a secure runtime that can be used as the initramfs for LinuxBoot. Take a look at the repository on GitHub. See osresearch.net for more documentation on HEADS.

Reply:
@tlaurion I couldn't find tinyboot. Is the original `ownerboot` [this project](https://git.newtype.fr/ryan/ownerboot)? And the only tvix I found was [a package manager](https://tvix.dev/).